tests/Request/VisionRequestTest.php

```diff
@@ -38,6 +48,66 @@ public function testPermissionsInjectedInPayload()
     }
 
     /**
+     *
+     */
+    public function testReturnsClientException()
+    {
+        $request = $this->getVisionRequest();
+
+        $class = new \ReflectionClass($request);
+        $property = $class->getProperty('clientException');
+        $property->setAccessible(true);
+        $property->setValue(
+        	$request,
+			new ClientException('exception message',
+				new Request('GET', '/'),
+				new Response(200,[],json_encode(["error" => ["code" => "test"]]))
+			)
+		);
+
+		$this->assertEquals($request->getAnnotateImageResponse()->getError()->getCode(), 'test');
+    }
+
+    /**
+     *
+     */
+    public function testReturnsAnnotateImageResponse()
+    {
+        $request = $this->getVisionRequest();
+
+        $class = new \ReflectionClass($request);
+        $property = $class->getProperty('rawResponse');
+        $property->setAccessible(true);
+        $property->setValue($request, json_encode(["responses" => [[]]]));
+
+        $this->assertTrue(get_class($request->getAnnotateImageResponse()) == AnnotateImageResponse::class);
```

Comment:
It's best to always do strict type statements (`===`) unless a method can return multiple return types.